From 7edda924a0ac962c98fac8342a474b6a2a057c1d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Ekl=C3=B6f?= <daniel@ekloef.se>
Date: Sat, 1 May 2021 22:24:12 +0200
Subject: [PATCH] =?UTF-8?q?meson:=20add=20=E2=80=98xkb=E2=80=99=20as=20a?=
 =?UTF-8?q?=20dependency=20to=20vtlib,=20pgolib=20and=20pgo?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

All of these include wayland.h (either directly, or indirectly), which
pulls in xkbcommon.h.

Reported by swayyyy on IRC:

  FAILED: libvtlib.a.p/dcs.c.o

  In file included from ../../terminal.h:19,
                   from ../../dcs.h:4,
                   from ../../dcs.c:1:
  ../../wayland.h:9:10: fatal error: xkbcommon/xkbcommon.h: No such file or directory
      9 | #include <xkbcommon/xkbcommon.h>
        |          ^
---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index cfdbe98d..370501f5 100644
--- a/meson.build
+++ b/meson.build
@@ -143,7 +143,7 @@ vtlib = static_library(
   'vt.c', 'vt.h',
   wl_proto_src + wl_proto_headers,
   version,
-  dependencies: [libepoll, pixman, fcft, tllist, wayland_client],
+  dependencies: [libepoll, pixman, fcft, tllist, wayland_client, xkb],
   link_with: [common, misc],
 )
 
@@ -153,7 +153,7 @@ pgolib = static_library(
   'selection.c', 'selection.h',
   'terminal.c', 'terminal.h',
   wl_proto_src + wl_proto_headers,
-  dependencies: [libepoll, pixman, fcft, tllist, wayland_client],
+  dependencies: [libepoll, pixman, fcft, tllist, wayland_client, xkb],
   link_with: vtlib,
 )
 
@@ -162,7 +162,7 @@ if get_option('b_pgo') == 'generate'
     'pgo',
     'pgo/pgo.c',
     wl_proto_src + wl_proto_headers,
-    dependencies: [math, threads, libepoll, pixman, wayland_client, fcft, tllist],
+    dependencies: [math, threads, libepoll, pixman, wayland_client, xkb, fcft, tllist],
     link_with: pgolib,
   )
 endif
